molo/core/views.py

```diff
@@ -130,18 +130,22 @@ def add_translation(request, page_id, locale):
     new_title = str(language) + " translation of %s" % page.title
     new_slug = generate_slug(new_title)
     translation = page.__class__(
-        title=new_title, slug=new_slug)
+        title=new_title, slug=new_slug, language=language)
     page.get_parent().add_child(instance=translation)
     translation.save_revision()
-    language_relation = translation.languages.first()
-    language_relation.language = language
-    language_relation.save()
-    translation.save_revision()
+    # add the translation the new way
+    page.specific.translated_pages.add(translation)
+    page.save()
+    for translated_page in \
+            page.specific.translated_pages.all():
+        translations = page.specific.translated_pages.all().\
+            exclude(language__pk=translated_page.language.pk)
+        for translation in translations:
+            translated_page.translated_pages.add(translation)
```

Comment:
if there are already some translated pages won't this add duplicates?

Reply:
@KaitCrawford this is in the add_translation method. So essentially this is a new page being created so no translations will be there yet